derivation.nix

```diff
@@ -15,15 +16,23 @@ let
 in
 {
   mkPnpmPackage =
-    { src
+    { workspace ? null
+    , components ? []
+    , src ? if (workspace != null && components != []) then workspace else null
     , packageJSON ? src + "/package.json"
+    , componentPackageJSONs ? map (c: {
+        name = "${c}/package.json";
+        value = src + "/${c}/package.json";
+      }) components
```

Comment:
Should log/fail here if these paths do not exist? My thinking is someone checks out marketplace and set ups direnv/nix prior to `git submodule update` which would mean `nucleus/packages.json` is missing. Not worthwhile if this would be tricky since we won't have the submodule for much longer.